### PR TITLE
UIImageView extension

### PR DESCRIPTION
**Pull request by RamonGilabert**
I guess it needs further testing in terms of the flashing of the images that we had, but here are some tests with Bad Internet connection and with WiFi:

bad internet

app

The flow was this one:

You create a task, you add it to the dictionary, it exists, so you cancel the task, this task completes and then it gives an error, in the closure, done in the background, it removes the value of the task from the dictionary, for some reason that was causing errors since this task was still going on or something, it seems to work, but I'd like further test.

### CODE REVIEW

File: Source/ImageFetcher.swift [hunk 1]
```diff
@@ -1,82 +1,74 @@
...
+          complete(imageAddress) { completion(result: .Failure(Error.InvalidStatusCode)) }
+          return
+        }
+
+        guard let data = data where httpResponse.validateLength(data) else {
+          complete(imageAddress) { completion(result: .Failure(Error.InvalidStatusCode)) }
+          return
+        }
+
+        guard let image = UIImage.decode(data) else {
```
Comment (vadymmarkov) [inline Source/ImageFetcher.swift:53]: I think decompression is on the main thread now :thought_balloon:

Comment (RamonGilabert) [inline Source/ImageFetcher.swift:53]: I checked that, the thing is that the task resolves in a random thread every time, so we are good.

Comment (vadymmarkov) [inline Source/ImageFetcher.swift:53]: But it's a main queue :scream:

**Comment by RamonGilabert** [comment]
Why that? :smile:

**Pull request closed without merging**